watchers/disruptions_watchers_manager.go

```diff
@@ -0,0 +1,217 @@
+}
+
+type WatcherName string
+
+const (
+	ChaosPodWatcherName         WatcherName = "ChaosPod"
+	DisruptionTargetWatcherName WatcherName = "DisruptionTarget"
+)
+
+var watchersNames = []WatcherName{
+	ChaosPodWatcherName,
+	DisruptionTargetWatcherName,
+}
+
+// CreateAllWatchers creates all the Watchers associated with the given Disruption.
+func (d disruptionsWatchersManager) CreateAllWatchers(disruption *v1beta1.Disruption, watcherManagerMock Manager, cacheMock k8scontrollercache.Cache) error {
+	// Check that the disruption object has a name and namespace
+	if disruption.ObjectMeta.Name == "" || disruption.ObjectMeta.Namespace == "" {
+		return fmt.Errorf("the disruption is not valid. It should contain a name and a namespace")
+	}
+
+	// Get the namespaced name of the disruption
+	disruptionNamespacedName := getDisruptionNamespacedName(disruption)
+
+	var watcherManager Manager
+
+	// If a mock watcher manager was passed in, use it
+	if watcherManagerMock == nil {
+		watcherManager = d.getWatcherManager(disruptionNamespacedName)
+	} else {
+		watcherManager = watcherManagerMock
+	}
+
+	// Save the watcher manager for later use
+	d.watchersManagers[disruptionNamespacedName] = watcherManager
+
+	// Calculate a hash of the disruption spec (excluding the count field)
+	disSpecHash, err := disruption.Spec.HashNoCount()
+	if err != nil {
+		return err
+	}
+
+	// For each type of watcher we need to create
+	for _, watcherName := range watchersNames {
+		// Calculate a hash for the watcher
+		watcherNameHash := disSpecHash + string(watcherName)
+
+		// If a watcher with this hash has already been created
+		if ok := watcherManager.GetWatcher(watcherNameHash); ok != nil {
+			continue
+		}
+
+		// Otherwise add the new watcher for the disruption
+		if err := d.addWatcher(disruption, watcherName, watcherNameHash, cacheMock, watcherManager); err != nil {
+			return err
+		}
+
+		d.log.Infof("Watcher %s created for disruption %s with %s namespace", watcherName, disruption.Name, disruption.Namespace)
+	}
+
+	return nil
+}
+
+// RemoveAllWatchers removes all the Watchers associated with the given Disruption.
+func (d disruptionsWatchersManager) RemoveAllWatchers(disruption *v1beta1.Disruption) {
+	namespacedName := getDisruptionNamespacedName(disruption)
+
+	// Get the Watcher Manager associated with the Disruption.
+	watcherManager := d.watchersManagers[namespacedName]
+
+	// If a Watcher Manager exists, remove all the Watchers associated with it.
+	if watcherManager != nil {
+		watcherManager.RemoveAllWatchers()
+
+		// Remove the Watcher Manager from the map.
+		delete(d.watchersManagers, namespacedName)
+
+		d.log.Infof("all watchers have been removed for the %s disruption with %s namespace", disruption.Name, disruption.Namespace)
+	}
+
+	d.log.Debugw(fmt.Sprintf("could not remove all watchers for the %s disruption with %s namespace", disruption.Name, disruption.Namespace), "disruption", disruption, "namespacedName", namespacedName)
```

Comment:
We don't want to dump the entire disruption struct as a log tag. 
```suggestion
	d.log.Debugw(fmt.Sprintf("could not remove all watchers for the %s disruption with %s namespace", disruption.Name, disruption.Namespace), "disruptionName", disruption.Name, "disruptionNamespace", namespacedName.Namespace)
```